src/Http/Paginators/PageRequestPaginator.php

```diff
@@ -0,0 +1,224 @@
+
+/**
+ * @template TRequest of \Saloon\Contracts\Request
+ * @template TResponse of \Saloon\Contracts\Response
+ *
+ * @extends RequestPaginator<TRequest, TResponse>
+ */
+class PageRequestPaginator extends RequestPaginator
+{
+    /**
+     * @var int
+     */
+    protected readonly int $originalPage;
+
+    /**
+     * @param \Saloon\Contracts\Connector $connector
+     * @param \Saloon\Contracts\Request $originalRequest
+     * @param int<0, max>|null $limit
+     * @param int<0, max> $page
+     */
+    public function __construct(
+        Connector $connector,
+        Request $originalRequest,
+        ?int $limit,
+        protected int $page = 1,
+    ) {
+        parent::__construct($connector, $originalRequest, $limit);
+
+        $this->originalPage = $page;
+    }
+
+    /**
+     * @return int
+     */
+    public function totalEntries(): int
+    {
+        // Make sure we have a response.
+        if (is_null($this->currentResponse)) {
+            $this->current();
+        }
+
+        return $this->currentResponse->json('total');
+    }
+
+    /**
+     * @return iterable<int, array<string, mixed>>
+     *
+     * @TODO entry data type
+     */
+    public function entries(): iterable
+    {
+        return $this->currentResponse->json('data');
+    }
+
+    /**
+     * @return int
+     */
+    public function totalPages(): int
+    {
+        // Make sure we have a response.
+        if (is_null($this->currentResponse)) {
+            $this->current();
+        }
+
+        return $this->lastPage();
+    }
+
+    /**
+     * @return int
+     */
+    public function firstPage(): int
+    {
+        // TODO: Or should we use something from the response?
+        return 1;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function previousPage(): ?int
+    {
+        return $this->currentPage() > $this->firstPage()
+            ? $this->currentPage() - 1
+            : null;
+    }
+
+    /**
+     * @return int
+     */
+    public function currentPage(): int
+    {
+        return $this->page;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function nextPage(): ?int
+    {
+        return $this->currentPage() < $this->totalPages()
+            ? $this->currentPage() + 1
+            : null;
+    }
+
+    /**
+     * @return int
+     */
+    public function lastPage(): int
+    {
+        return $this->currentResponse->json('last_page');
+    }
+
+    protected function applyPaging(Request $request): void
+    {
+        if (! is_null($this->limit())) {
+            $request->query()->add('limit', $this->limit());
+        }
+
+        $request->query()->add('page', $this->currentPage());
+    }
+
+    /**
+     * @return void
+     */
+    public function rewind(): void
+    {
+        if (! $this->shouldRewind()) {
+            return;
+        }
+
+        parent::rewind();
+
+        // TODO: Rewind completely, or rewind to originalPage?
+        $this->page = 1;
```

Comment:
Original page I reckon

Reply:
Yeah, that's what I'm thinking.
If we were to continue the iteration somewhere, and original offset is not 0/the lowest offset.
_Rewinding_ should be to rewind to the start, imo.
But it could also give unpredictable behaviour, if the first position (original offset/page) is not the lowest, then we'd still send request to the lowest..